lib/core/src/recover/model.rs

```diff
@@ -168,25 +196,41 @@ pub(crate) struct RecoveredOnchainDataChainReceive {
     /// BTC tx initiated by the SDK to a user-chosen address, in case the initial funds have to be refunded.
     pub(crate) btc_refund_tx_id: Option<HistoryTxId>,
 }
+// TODO: We have to be careful around overwriting the Refundable or RefundPending state, as this swap monitored
+// after the expiration of the swap and if new funds are detected on the lockup script they are either refunded
+// or marked refundable. Perhaps we should check in the recovery the lockup balance and set accordingly.
 impl PartialSwapState for RecoveredOnchainDataChainReceive {
-    fn derive_partial_state(&self) -> Option<PaymentState> {
+    fn derive_partial_state(&self, is_expired: bool) -> Option<PaymentState> {
         match &self.btc_user_lockup_tx_id {
-            Some(_) => match &self.lbtc_claim_tx_id {
-                Some(lbtc_claim_tx_id) => match lbtc_claim_tx_id.confirmed() {
+            Some(_) => match (&self.lbtc_claim_tx_id, &self.btc_refund_tx_id) {
+                (Some(lbtc_claim_tx_id), None) => match lbtc_claim_tx_id.confirmed() {
                     true => Some(PaymentState::Complete),
                     false => Some(PaymentState::Pending),
                 },
-                None => match &self.btc_refund_tx_id {
-                    Some(tx) => match tx.confirmed() {
-                        true => Some(PaymentState::Failed),
+                (None, Some(btc_refund_tx_id)) => match btc_refund_tx_id.confirmed() {
+                    true => Some(PaymentState::Failed),
+                    false => Some(PaymentState::RefundPending),
+                },
+                (Some(lbtc_claim_tx_id), Some(btc_refund_tx_id)) => {
+                    match btc_refund_tx_id.confirmed() {
+                        true => match lbtc_claim_tx_id.confirmed() {
+                            true => Some(PaymentState::Complete),
+                            false => Some(PaymentState::Pending),
+                        },
                         false => Some(PaymentState::RefundPending),
-                    },
-                    None => Some(PaymentState::Pending),
+                    }
```

Comment:
A bit of a weird scenario here. Shouldn't the claim tx take precedence over the refund tx and set the state that way? So:
- Claim Confirmed, Refund (any state) => `Complete`
- Claim Not Confirmed, Refund Confirmed  => `Failed`
- Claim Not Confirmed, Refund Not Confirmed => `RefundPending` (only scenario where this can happen is if Boltz broadcasts a claim which doesn't get confirmed, the block height timeout expires and we broadcast a refund. In this case IMO the more proper state would be RefundPending)

Edit: This can technically never happen given this line: https://github.com/breez/breez-sdk-liquid/blob/e4e1ed95c1775b32676c83afc5d5ec43bef2ac5a/lib/core/src/recover/recoverer.rs#L628-L633
but good to have anyway.

Reply:
It's more to handle the address reuse case, where there could be a claim and refund(s). But yeah, it's unlikely that the claim unconfirmed while a refund is unconfirmed.

This `btc_refund_tx_id ` handling is not right btw, it should always return the latest refund tx id if there is one. I'm making that change now